cwms-data-api/src/main/java/cwms/cda/data/dao/LocationLevelsDaoImpl.java

```diff
@@ -330,6 +331,11 @@ public LocationLevel retrieveLocationLevel(String locationLevelName, String pUni
         return connectionResult(dsl, c -> {
             String units = pUnits;
             Configuration configuration = getDslContext(c, officeId).configuration();
+            if (units != null && (units.equalsIgnoreCase("SI")
+                    || units.equalsIgnoreCase("EN"))) {
+                units = CWMS_UTIL_PACKAGE.call_GET_DEFAULT_UNITS(configuration,
+                        locationLevelName.split("\\.")[1], units);
```

Comment:
I don't see validation on the location level id prior to this, the incoming string might not have any periods and would then throw and ArrayIndexOutOfBoundsException

Reply:
Added a format check before calling methods using the split string

Reply:
Added a pattern and cleaned up usage